[2025-05-21T09:30:00.797+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: yellow_taxi_batch_pipeline.transform_taxi_data manual__2025-05-21T09:27:24.603467+00:00 [queued]>
[2025-05-21T09:30:00.815+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: yellow_taxi_batch_pipeline.transform_taxi_data manual__2025-05-21T09:27:24.603467+00:00 [queued]>
[2025-05-21T09:30:00.816+0000] {taskinstance.py:1359} INFO - Starting attempt 1 of 2
[2025-05-21T09:30:00.851+0000] {taskinstance.py:1380} INFO - Executing <Task(BashOperator): transform_taxi_data> on 2025-05-21 09:27:24.603467+00:00
[2025-05-21T09:30:00.859+0000] {standard_task_runner.py:57} INFO - Started process 789 to run task
[2025-05-21T09:30:00.864+0000] {standard_task_runner.py:84} INFO - Running: ['airflow', 'tasks', 'run', 'yellow_taxi_batch_pipeline', 'transform_taxi_data', 'manual__2025-05-21T09:27:24.603467+00:00', '--job-id', '63', '--raw', '--subdir', 'DAGS_FOLDER/taxi_batch_dag.py', '--cfg-path', '/tmp/tmp593yexxx']
[2025-05-21T09:30:00.868+0000] {standard_task_runner.py:85} INFO - Job 63: Subtask transform_taxi_data
[2025-05-21T09:30:00.920+0000] {logging_mixin.py:151} WARNING - /home/airflow/.local/lib/python3.8/site-packages/airflow/settings.py:195 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
[2025-05-21T09:30:00.994+0000] {task_command.py:415} INFO - Running <TaskInstance: yellow_taxi_batch_pipeline.transform_taxi_data manual__2025-05-21T09:27:24.603467+00:00 [running]> on host c4c55d29c79f
[2025-05-21T09:30:01.233+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='yellow_taxi_batch_pipeline' AIRFLOW_CTX_TASK_ID='transform_taxi_data' AIRFLOW_CTX_EXECUTION_DATE='2025-05-21T09:27:24.603467+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-05-21T09:27:24.603467+00:00'
[2025-05-21T09:30:01.244+0000] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2025-05-21T09:30:01.246+0000] {subprocess.py:75} INFO - Running command: ['/bin/bash', '-c', '\n    spark-submit         --master spark://spark-master:7077         --jars /opt/airflow/dags/jars/***ql-42.5.0.jar         --packages org.apache.hadoop:hadoop-aws:3.3.1         /opt/airflow/dags/scripts/taxi_transform.py\n    ']
[2025-05-21T09:30:01.295+0000] {subprocess.py:86} INFO - Output:
[2025-05-21T09:30:01.318+0000] {subprocess.py:93} INFO - /opt/spark/bin/load-spark-env.sh: line 68: ps: command not found
[2025-05-21T09:30:05.315+0000] {subprocess.py:93} INFO - :: loading settings :: url = jar:file:/opt/spark/jars/ivy-2.5.0.jar!/org/apache/ivy/core/settings/ivysettings.xml
[2025-05-21T09:30:05.532+0000] {subprocess.py:93} INFO - Ivy Default Cache set to: /home/airflow/.ivy2/cache
[2025-05-21T09:30:05.534+0000] {subprocess.py:93} INFO - The jars for the packages stored in: /home/airflow/.ivy2/jars
[2025-05-21T09:30:05.553+0000] {subprocess.py:93} INFO - org.apache.hadoop#hadoop-aws added as a dependency
[2025-05-21T09:30:05.555+0000] {subprocess.py:93} INFO - :: resolving dependencies :: org.apache.spark#spark-submit-parent-9d71968f-d0b0-4163-af0b-9dd29ada06ec;1.0
[2025-05-21T09:30:05.557+0000] {subprocess.py:93} INFO - 	confs: [default]
[2025-05-21T09:30:06.129+0000] {subprocess.py:93} INFO - 	found org.apache.hadoop#hadoop-aws;3.3.1 in central
[2025-05-21T09:30:06.215+0000] {subprocess.py:93} INFO - 	found com.amazonaws#aws-java-sdk-bundle;1.11.901 in central
[2025-05-21T09:30:06.297+0000] {subprocess.py:93} INFO - 	found org.wildfly.openssl#wildfly-openssl;1.0.7.Final in central
[2025-05-21T09:30:06.382+0000] {subprocess.py:93} INFO - :: resolution report :: resolve 795ms :: artifacts dl 33ms
[2025-05-21T09:30:06.383+0000] {subprocess.py:93} INFO - 	:: modules in use:
[2025-05-21T09:30:06.384+0000] {subprocess.py:93} INFO - 	com.amazonaws#aws-java-sdk-bundle;1.11.901 from central in [default]
[2025-05-21T09:30:06.385+0000] {subprocess.py:93} INFO - 	org.apache.hadoop#hadoop-aws;3.3.1 from central in [default]
[2025-05-21T09:30:06.409+0000] {subprocess.py:93} INFO - 	org.wildfly.openssl#wildfly-openssl;1.0.7.Final from central in [default]
[2025-05-21T09:30:06.410+0000] {subprocess.py:93} INFO - 	---------------------------------------------------------------------
[2025-05-21T09:30:06.415+0000] {subprocess.py:93} INFO - 	|                  |            modules            ||   artifacts   |
[2025-05-21T09:30:06.416+0000] {subprocess.py:93} INFO - 	|       conf       | number| search|dwnlded|evicted|| number|dwnlded|
[2025-05-21T09:30:06.417+0000] {subprocess.py:93} INFO - 	---------------------------------------------------------------------
[2025-05-21T09:30:06.417+0000] {subprocess.py:93} INFO - 	|      default     |   3   |   0   |   0   |   0   ||   3   |   0   |
[2025-05-21T09:30:06.418+0000] {subprocess.py:93} INFO - 	---------------------------------------------------------------------
[2025-05-21T09:30:06.418+0000] {subprocess.py:93} INFO - :: retrieving :: org.apache.spark#spark-submit-parent-9d71968f-d0b0-4163-af0b-9dd29ada06ec
[2025-05-21T09:30:06.419+0000] {subprocess.py:93} INFO - 	confs: [default]
[2025-05-21T09:30:06.446+0000] {subprocess.py:93} INFO - 	0 artifacts copied, 3 already retrieved (0kB/32ms)
[2025-05-21T09:30:07.504+0000] {subprocess.py:93} INFO - 25/05/21 09:30:07 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2025-05-21T09:30:07.972+0000] {subprocess.py:93} INFO - 25/05/21 09:30:07 WARN DependencyUtils: Local jar /opt/airflow/dags/jars/***ql-42.5.0.jar does not exist, skipping.
[2025-05-21T09:30:10.892+0000] {subprocess.py:93} INFO - 25/05/21 09:30:10 INFO SparkContext: Running Spark version 3.3.1
[2025-05-21T09:30:10.977+0000] {subprocess.py:93} INFO - 25/05/21 09:30:10 INFO ResourceUtils: ==============================================================
[2025-05-21T09:30:10.978+0000] {subprocess.py:93} INFO - 25/05/21 09:30:10 INFO ResourceUtils: No custom resources configured for spark.driver.
[2025-05-21T09:30:10.980+0000] {subprocess.py:93} INFO - 25/05/21 09:30:10 INFO ResourceUtils: ==============================================================
[2025-05-21T09:30:10.981+0000] {subprocess.py:93} INFO - 25/05/21 09:30:10 INFO SparkContext: Submitted application: YellowTaxiTransform
[2025-05-21T09:30:11.046+0000] {subprocess.py:93} INFO - 25/05/21 09:30:11 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(cores -> name: cores, amount: 1, script: , vendor: , memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2025-05-21T09:30:11.086+0000] {subprocess.py:93} INFO - 25/05/21 09:30:11 INFO ResourceProfile: Limiting resource is cpu
[2025-05-21T09:30:11.095+0000] {subprocess.py:93} INFO - 25/05/21 09:30:11 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2025-05-21T09:30:11.269+0000] {subprocess.py:93} INFO - 25/05/21 09:30:11 INFO SecurityManager: Changing view acls to: airflow
[2025-05-21T09:30:11.271+0000] {subprocess.py:93} INFO - 25/05/21 09:30:11 INFO SecurityManager: Changing modify acls to: airflow
[2025-05-21T09:30:11.274+0000] {subprocess.py:93} INFO - 25/05/21 09:30:11 INFO SecurityManager: Changing view acls groups to:
[2025-05-21T09:30:11.277+0000] {subprocess.py:93} INFO - 25/05/21 09:30:11 INFO SecurityManager: Changing modify acls groups to:
[2025-05-21T09:30:11.279+0000] {subprocess.py:93} INFO - 25/05/21 09:30:11 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users  with view permissions: Set(airflow); groups with view permissions: Set(); users  with modify permissions: Set(airflow); groups with modify permissions: Set()
[2025-05-21T09:30:12.119+0000] {subprocess.py:93} INFO - 25/05/21 09:30:12 INFO Utils: Successfully started service 'sparkDriver' on port 42461.
[2025-05-21T09:30:12.229+0000] {subprocess.py:93} INFO - 25/05/21 09:30:12 INFO SparkEnv: Registering MapOutputTracker
[2025-05-21T09:30:12.317+0000] {subprocess.py:93} INFO - 25/05/21 09:30:12 INFO SparkEnv: Registering BlockManagerMaster
[2025-05-21T09:30:12.377+0000] {subprocess.py:93} INFO - 25/05/21 09:30:12 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2025-05-21T09:30:12.388+0000] {subprocess.py:93} INFO - 25/05/21 09:30:12 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2025-05-21T09:30:12.390+0000] {subprocess.py:93} INFO - 25/05/21 09:30:12 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2025-05-21T09:30:12.467+0000] {subprocess.py:93} INFO - 25/05/21 09:30:12 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-90491f37-33cd-4084-8117-1f70af2fb7e9
[2025-05-21T09:30:12.520+0000] {subprocess.py:93} INFO - 25/05/21 09:30:12 INFO MemoryStore: MemoryStore started with capacity 434.4 MiB
[2025-05-21T09:30:12.589+0000] {subprocess.py:93} INFO - 25/05/21 09:30:12 INFO SparkEnv: Registering OutputCommitCoordinator
[2025-05-21T09:30:13.144+0000] {subprocess.py:93} INFO - 25/05/21 09:30:13 WARN Utils: Service 'SparkUI' could not bind on port 4040. Attempting port 4041.
[2025-05-21T09:30:13.177+0000] {subprocess.py:93} INFO - 25/05/21 09:30:13 INFO Utils: Successfully started service 'SparkUI' on port 4041.
[2025-05-21T09:30:13.305+0000] {subprocess.py:93} INFO - 25/05/21 09:30:13 ERROR SparkContext: Failed to add /opt/spark/jars/***ql-42.5.0.jar to Spark environment
[2025-05-21T09:30:13.306+0000] {subprocess.py:93} INFO - java.io.FileNotFoundException: Jar /opt/spark/jars/***ql-42.5.0.jar not found
[2025-05-21T09:30:13.307+0000] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.addLocalJarFile$1(SparkContext.scala:1949)
[2025-05-21T09:30:13.308+0000] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.addJar(SparkContext.scala:2004)
[2025-05-21T09:30:13.309+0000] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.$anonfun$new$12(SparkContext.scala:507)
[2025-05-21T09:30:13.309+0000] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.$anonfun$new$12$adapted(SparkContext.scala:507)
[2025-05-21T09:30:13.310+0000] {subprocess.py:93} INFO - 	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
[2025-05-21T09:30:13.311+0000] {subprocess.py:93} INFO - 	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
[2025-05-21T09:30:13.312+0000] {subprocess.py:93} INFO - 	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
[2025-05-21T09:30:13.312+0000] {subprocess.py:93} INFO - 	at org.apache.spark.SparkContext.<init>(SparkContext.scala:507)
[2025-05-21T09:30:13.313+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.java.JavaSparkContext.<init>(JavaSparkContext.scala:58)
[2025-05-21T09:30:13.314+0000] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[2025-05-21T09:30:13.314+0000] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
[2025-05-21T09:30:13.315+0000] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[2025-05-21T09:30:13.315+0000] {subprocess.py:93} INFO - 	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
[2025-05-21T09:30:13.316+0000] {subprocess.py:93} INFO - 	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:247)
[2025-05-21T09:30:13.321+0000] {subprocess.py:93} INFO - 	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
[2025-05-21T09:30:13.322+0000] {subprocess.py:93} INFO - 	at py4j.Gateway.invoke(Gateway.java:238)
[2025-05-21T09:30:13.326+0000] {subprocess.py:93} INFO - 	at py4j.commands.ConstructorCommand.invokeConstructor(ConstructorCommand.java:80)
[2025-05-21T09:30:13.327+0000] {subprocess.py:93} INFO - 	at py4j.commands.ConstructorCommand.execute(ConstructorCommand.java:69)
[2025-05-21T09:30:13.328+0000] {subprocess.py:93} INFO - 	at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2025-05-21T09:30:13.329+0000] {subprocess.py:93} INFO - 	at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2025-05-21T09:30:13.329+0000] {subprocess.py:93} INFO - 	at java.base/java.lang.Thread.run(Thread.java:829)
[2025-05-21T09:30:13.330+0000] {subprocess.py:93} INFO - 25/05/21 09:30:13 INFO SparkContext: Added file file:///home/airflow/.ivy2/jars/org.apache.hadoop_hadoop-aws-3.3.1.jar at spark://c4c55d29c79f:42461/files/org.apache.hadoop_hadoop-aws-3.3.1.jar with timestamp 1747819810860
[2025-05-21T09:30:13.331+0000] {subprocess.py:93} INFO - 25/05/21 09:30:13 INFO Utils: Copying /home/airflow/.ivy2/jars/org.apache.hadoop_hadoop-aws-3.3.1.jar to /tmp/spark-39f06296-48ba-446f-ae64-ef1f9fabfa4e/userFiles-82762fd8-9a3c-4e5b-b9db-e3d90f5e6918/org.apache.hadoop_hadoop-aws-3.3.1.jar
[2025-05-21T09:30:13.367+0000] {subprocess.py:93} INFO - 25/05/21 09:30:13 INFO SparkContext: Added file file:///home/airflow/.ivy2/jars/com.amazonaws_aws-java-sdk-bundle-1.11.901.jar at spark://c4c55d29c79f:42461/files/com.amazonaws_aws-java-sdk-bundle-1.11.901.jar with timestamp 1747819810860
[2025-05-21T09:30:13.368+0000] {subprocess.py:93} INFO - 25/05/21 09:30:13 INFO Utils: Copying /home/airflow/.ivy2/jars/com.amazonaws_aws-java-sdk-bundle-1.11.901.jar to /tmp/spark-39f06296-48ba-446f-ae64-ef1f9fabfa4e/userFiles-82762fd8-9a3c-4e5b-b9db-e3d90f5e6918/com.amazonaws_aws-java-sdk-bundle-1.11.901.jar
[2025-05-21T09:30:16.282+0000] {subprocess.py:93} INFO - 25/05/21 09:30:16 INFO SparkContext: Added file file:///home/airflow/.ivy2/jars/org.wildfly.openssl_wildfly-openssl-1.0.7.Final.jar at spark://c4c55d29c79f:42461/files/org.wildfly.openssl_wildfly-openssl-1.0.7.Final.jar with timestamp 1747819810860
[2025-05-21T09:30:16.286+0000] {subprocess.py:93} INFO - 25/05/21 09:30:16 INFO Utils: Copying /home/airflow/.ivy2/jars/org.wildfly.openssl_wildfly-openssl-1.0.7.Final.jar to /tmp/spark-39f06296-48ba-446f-ae64-ef1f9fabfa4e/userFiles-82762fd8-9a3c-4e5b-b9db-e3d90f5e6918/org.wildfly.openssl_wildfly-openssl-1.0.7.Final.jar
[2025-05-21T09:30:16.621+0000] {subprocess.py:93} INFO - 25/05/21 09:30:16 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://spark-master:7077...
[2025-05-21T09:30:16.790+0000] {subprocess.py:93} INFO - 25/05/21 09:30:16 INFO TransportClientFactory: Successfully created connection to spark-master/172.18.0.3:7077 after 112 ms (0 ms spent in bootstraps)
[2025-05-21T09:30:17.066+0000] {subprocess.py:93} INFO - 25/05/21 09:30:17 INFO StandaloneSchedulerBackend: Connected to Spark cluster with app ID app-20250521093017-0002
[2025-05-21T09:30:17.086+0000] {subprocess.py:93} INFO - 25/05/21 09:30:17 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 39563.
[2025-05-21T09:30:17.087+0000] {subprocess.py:93} INFO - 25/05/21 09:30:17 INFO NettyBlockTransferService: Server created on c4c55d29c79f:39563
[2025-05-21T09:30:17.091+0000] {subprocess.py:93} INFO - 25/05/21 09:30:17 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2025-05-21T09:30:17.107+0000] {subprocess.py:93} INFO - 25/05/21 09:30:17 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, c4c55d29c79f, 39563, None)
[2025-05-21T09:30:17.113+0000] {subprocess.py:93} INFO - 25/05/21 09:30:17 INFO BlockManagerMasterEndpoint: Registering block manager c4c55d29c79f:39563 with 434.4 MiB RAM, BlockManagerId(driver, c4c55d29c79f, 39563, None)
[2025-05-21T09:30:17.119+0000] {subprocess.py:93} INFO - 25/05/21 09:30:17 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, c4c55d29c79f, 39563, None)
[2025-05-21T09:30:17.123+0000] {subprocess.py:93} INFO - 25/05/21 09:30:17 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, c4c55d29c79f, 39563, None)
[2025-05-21T09:30:17.670+0000] {subprocess.py:93} INFO - 25/05/21 09:30:17 INFO StandaloneAppClient$ClientEndpoint: Executor added: app-20250521093017-0002/0 on worker-20250521092629-172.18.0.5-43253 (172.18.0.5:43253) with 2 core(s)
[2025-05-21T09:30:17.674+0000] {subprocess.py:93} INFO - 25/05/21 09:30:17 INFO StandaloneSchedulerBackend: Granted executor ID app-20250521093017-0002/0 on hostPort 172.18.0.5:43253 with 2 core(s), 1024.0 MiB RAM
[2025-05-21T09:30:17.707+0000] {subprocess.py:93} INFO - 25/05/21 09:30:17 INFO StandaloneSchedulerBackend: SchedulerBackend is ready for scheduling beginning after reached minRegisteredResourcesRatio: 0.0
[2025-05-21T09:30:17.853+0000] {subprocess.py:93} INFO - 25/05/21 09:30:17 INFO StandaloneAppClient$ClientEndpoint: Executor updated: app-20250521093017-0002/0 is now RUNNING
[2025-05-21T09:30:18.174+0000] {subprocess.py:93} INFO - 25/05/21 09:30:18 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir.
[2025-05-21T09:30:18.179+0000] {subprocess.py:93} INFO - 25/05/21 09:30:18 INFO SharedState: Warehouse path is 'file:/tmp/airflowtmptbcqbxpf/spark-warehouse'.
[2025-05-21T09:30:20.306+0000] {subprocess.py:93} INFO - 2025-05-21 09:30:20,305 - INFO - Loading Yellow Taxi data from MinIO
[2025-05-21T09:30:20.699+0000] {subprocess.py:93} INFO - 25/05/21 09:30:20 WARN MetricsConfig: Cannot locate configuration: tried hadoop-metrics2-s3a-file-system.properties,hadoop-metrics2.properties
[2025-05-21T09:30:20.720+0000] {subprocess.py:93} INFO - 25/05/21 09:30:20 INFO MetricsSystemImpl: Scheduled Metric snapshot period at 10 second(s).
[2025-05-21T09:30:20.721+0000] {subprocess.py:93} INFO - 25/05/21 09:30:20 INFO MetricsSystemImpl: s3a-file-system metrics system started
[2025-05-21T09:30:24.417+0000] {subprocess.py:93} INFO - 25/05/21 09:30:24 INFO CoarseGrainedSchedulerBackend$DriverEndpoint: Registered executor NettyRpcEndpointRef(spark-client://Executor) (172.18.0.5:44982) with ID 0,  ResourceProfileId 0
[2025-05-21T09:30:24.535+0000] {subprocess.py:93} INFO - 25/05/21 09:30:24 INFO BlockManagerMasterEndpoint: Registering block manager 172.18.0.5:40317 with 366.3 MiB RAM, BlockManagerId(0, 172.18.0.5, 40317, None)
[2025-05-21T09:30:24.605+0000] {subprocess.py:93} INFO - 25/05/21 09:30:24 WARN FileStreamSink: Assume no metadata directory. Error while looking for metadata directory in the path: s3a://nyc-taxi-data/yellow_taxi/yellow_tripdata_2022-*.parquet.
[2025-05-21T09:30:24.606+0000] {subprocess.py:93} INFO - java.nio.file.AccessDeniedException: s3a://nyc-taxi-data/yellow_taxi/yellow_tripdata_2022-*.parquet: org.apache.hadoop.fs.s3a.auth.NoAuthWithAWSException: No AWS Credentials provided by TemporaryAWSCredentialsProvider SimpleAWSCredentialsProvider EnvironmentVariableCredentialsProvider IAMInstanceCredentialsProvider : com.amazonaws.SdkClientException: Unable to load AWS credentials from environment variables (AWS_ACCESS_KEY_ID (or AWS_ACCESS_KEY) and AWS_SECRET_KEY (or AWS_SECRET_ACCESS_KEY))
[2025-05-21T09:30:24.607+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AUtils.translateException(S3AUtils.java:206)
[2025-05-21T09:30:24.608+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AUtils.translateException(S3AUtils.java:170)
[2025-05-21T09:30:24.608+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AFileSystem.s3GetFileStatus(S3AFileSystem.java:3351)
[2025-05-21T09:30:24.608+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AFileSystem.innerGetFileStatus(S3AFileSystem.java:3185)
[2025-05-21T09:30:24.609+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AFileSystem.isDirectory(S3AFileSystem.java:4277)
[2025-05-21T09:30:24.610+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.FileStreamSink$.hasMetadata(FileStreamSink.scala:54)
[2025-05-21T09:30:24.611+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.DataSource.resolveRelation(DataSource.scala:370)
[2025-05-21T09:30:24.611+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.DataFrameReader.loadV1Source(DataFrameReader.scala:228)
[2025-05-21T09:30:24.612+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.DataFrameReader.$anonfun$load$2(DataFrameReader.scala:210)
[2025-05-21T09:30:24.612+0000] {subprocess.py:93} INFO - 	at scala.Option.getOrElse(Option.scala:189)
[2025-05-21T09:30:24.613+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:210)
[2025-05-21T09:30:24.613+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.DataFrameReader.parquet(DataFrameReader.scala:562)
[2025-05-21T09:30:24.613+0000] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2025-05-21T09:30:24.614+0000] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[2025-05-21T09:30:24.614+0000] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2025-05-21T09:30:24.614+0000] {subprocess.py:93} INFO - 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[2025-05-21T09:30:24.616+0000] {subprocess.py:93} INFO - 	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
[2025-05-21T09:30:24.619+0000] {subprocess.py:93} INFO - 	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
[2025-05-21T09:30:24.621+0000] {subprocess.py:93} INFO - 	at py4j.Gateway.invoke(Gateway.java:282)
[2025-05-21T09:30:24.629+0000] {subprocess.py:93} INFO - 	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
[2025-05-21T09:30:24.637+0000] {subprocess.py:93} INFO - 	at py4j.commands.CallCommand.execute(CallCommand.java:79)
[2025-05-21T09:30:24.638+0000] {subprocess.py:93} INFO - 	at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2025-05-21T09:30:24.641+0000] {subprocess.py:93} INFO - 	at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2025-05-21T09:30:24.653+0000] {subprocess.py:93} INFO - 	at java.base/java.lang.Thread.run(Thread.java:829)
[2025-05-21T09:30:24.655+0000] {subprocess.py:93} INFO - Caused by: org.apache.hadoop.fs.s3a.auth.NoAuthWithAWSException: No AWS Credentials provided by TemporaryAWSCredentialsProvider SimpleAWSCredentialsProvider EnvironmentVariableCredentialsProvider IAMInstanceCredentialsProvider : com.amazonaws.SdkClientException: Unable to load AWS credentials from environment variables (AWS_ACCESS_KEY_ID (or AWS_ACCESS_KEY) and AWS_SECRET_KEY (or AWS_SECRET_ACCESS_KEY))
[2025-05-21T09:30:24.657+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.AWSCredentialProviderList.getCredentials(AWSCredentialProviderList.java:216)
[2025-05-21T09:30:24.658+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.getCredentialsFromContext(AmazonHttpClient.java:1257)
[2025-05-21T09:30:24.659+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.runBeforeRequestHandlers(AmazonHttpClient.java:833)
[2025-05-21T09:30:24.660+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:783)
[2025-05-21T09:30:24.660+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:770)
[2025-05-21T09:30:24.661+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:744)
[2025-05-21T09:30:24.661+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:704)
[2025-05-21T09:30:24.661+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:686)
[2025-05-21T09:30:24.662+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:550)
[2025-05-21T09:30:24.662+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:530)
[2025-05-21T09:30:24.663+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5227)
[2025-05-21T09:30:24.663+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.getBucketRegionViaHeadRequest(AmazonS3Client.java:6189)
[2025-05-21T09:30:24.664+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.fetchRegionFromCache(AmazonS3Client.java:6162)
[2025-05-21T09:30:24.664+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5211)
[2025-05-21T09:30:24.665+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5173)
[2025-05-21T09:30:24.666+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5167)
[2025-05-21T09:30:24.666+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.listObjectsV2(AmazonS3Client.java:963)
[2025-05-21T09:30:24.667+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AFileSystem.lambda$listObjects$7(S3AFileSystem.java:2116)
[2025-05-21T09:30:24.667+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.statistics.impl.IOStatisticsBinding.lambda$trackDurationOfOperation$5(IOStatisticsBinding.java:499)
[2025-05-21T09:30:24.670+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.Invoker.retryUntranslated(Invoker.java:412)
[2025-05-21T09:30:24.671+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.Invoker.retryUntranslated(Invoker.java:375)
[2025-05-21T09:30:24.672+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AFileSystem.listObjects(S3AFileSystem.java:2107)
[2025-05-21T09:30:24.672+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AFileSystem.s3GetFileStatus(S3AFileSystem.java:3322)
[2025-05-21T09:30:24.673+0000] {subprocess.py:93} INFO - 	... 21 more
[2025-05-21T09:30:24.673+0000] {subprocess.py:93} INFO - Caused by: com.amazonaws.SdkClientException: Unable to load AWS credentials from environment variables (AWS_ACCESS_KEY_ID (or AWS_ACCESS_KEY) and AWS_SECRET_KEY (or AWS_SECRET_ACCESS_KEY))
[2025-05-21T09:30:24.674+0000] {subprocess.py:93} INFO - 	at com.amazonaws.auth.EnvironmentVariableCredentialsProvider.getCredentials(EnvironmentVariableCredentialsProvider.java:50)
[2025-05-21T09:30:24.675+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.AWSCredentialProviderList.getCredentials(AWSCredentialProviderList.java:177)
[2025-05-21T09:30:24.675+0000] {subprocess.py:93} INFO - 	... 43 more
[2025-05-21T09:30:26.789+0000] {subprocess.py:93} INFO - 2025-05-21 09:30:26,789 - ERROR - ETL process failed: An error occurred while calling o31.parquet.
[2025-05-21T09:30:26.790+0000] {subprocess.py:93} INFO - : java.nio.file.AccessDeniedException: s3a://nyc-taxi-data/yellow_taxi: org.apache.hadoop.fs.s3a.auth.NoAuthWithAWSException: No AWS Credentials provided by TemporaryAWSCredentialsProvider SimpleAWSCredentialsProvider EnvironmentVariableCredentialsProvider IAMInstanceCredentialsProvider : com.amazonaws.SdkClientException: Unable to load AWS credentials from environment variables (AWS_ACCESS_KEY_ID (or AWS_ACCESS_KEY) and AWS_SECRET_KEY (or AWS_SECRET_ACCESS_KEY))
[2025-05-21T09:30:26.791+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AUtils.translateException(S3AUtils.java:206)
[2025-05-21T09:30:26.791+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.Invoker.once(Invoker.java:117)
[2025-05-21T09:30:26.792+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AFileSystem.listStatus(S3AFileSystem.java:2810)
[2025-05-21T09:30:26.792+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.Globber.listStatus(Globber.java:128)
[2025-05-21T09:30:26.793+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.Globber.doGlob(Globber.java:291)
[2025-05-21T09:30:26.793+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.Globber.glob(Globber.java:202)
[2025-05-21T09:30:26.793+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AFileSystem.globStatus(S3AFileSystem.java:4253)
[2025-05-21T09:30:26.794+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AFileSystem.globStatus(S3AFileSystem.java:4233)
[2025-05-21T09:30:26.794+0000] {subprocess.py:93} INFO - 	at org.apache.spark.deploy.SparkHadoopUtil.globPath(SparkHadoopUtil.scala:253)
[2025-05-21T09:30:26.795+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.DataSource$.$anonfun$checkAndGlobPathIfNecessary$3(DataSource.scala:765)
[2025-05-21T09:30:26.795+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.ThreadUtils$.$anonfun$parmap$2(ThreadUtils.scala:372)
[2025-05-21T09:30:26.795+0000] {subprocess.py:93} INFO - 	at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:659)
[2025-05-21T09:30:26.796+0000] {subprocess.py:93} INFO - 	at scala.util.Success.$anonfun$map$1(Try.scala:255)
[2025-05-21T09:30:26.796+0000] {subprocess.py:93} INFO - 	at scala.util.Success.map(Try.scala:213)
[2025-05-21T09:30:26.797+0000] {subprocess.py:93} INFO - 	at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
[2025-05-21T09:30:26.797+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
[2025-05-21T09:30:26.798+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
[2025-05-21T09:30:26.798+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
[2025-05-21T09:30:26.799+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1426)
[2025-05-21T09:30:26.799+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
[2025-05-21T09:30:26.799+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
[2025-05-21T09:30:26.800+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
[2025-05-21T09:30:26.800+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
[2025-05-21T09:30:26.801+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
[2025-05-21T09:30:26.801+0000] {subprocess.py:93} INFO - Caused by: org.apache.hadoop.fs.s3a.auth.NoAuthWithAWSException: No AWS Credentials provided by TemporaryAWSCredentialsProvider SimpleAWSCredentialsProvider EnvironmentVariableCredentialsProvider IAMInstanceCredentialsProvider : com.amazonaws.SdkClientException: Unable to load AWS credentials from environment variables (AWS_ACCESS_KEY_ID (or AWS_ACCESS_KEY) and AWS_SECRET_KEY (or AWS_SECRET_ACCESS_KEY))
[2025-05-21T09:30:26.801+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.AWSCredentialProviderList.getCredentials(AWSCredentialProviderList.java:216)
[2025-05-21T09:30:26.802+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.getCredentialsFromContext(AmazonHttpClient.java:1257)
[2025-05-21T09:30:26.802+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.runBeforeRequestHandlers(AmazonHttpClient.java:833)
[2025-05-21T09:30:26.803+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:783)
[2025-05-21T09:30:26.803+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:770)
[2025-05-21T09:30:26.803+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:744)
[2025-05-21T09:30:26.804+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:704)
[2025-05-21T09:30:26.804+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:686)
[2025-05-21T09:30:26.805+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:550)
[2025-05-21T09:30:26.806+0000] {subprocess.py:93} INFO - 	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:530)
[2025-05-21T09:30:26.806+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5227)
[2025-05-21T09:30:26.807+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.getBucketRegionViaHeadRequest(AmazonS3Client.java:6189)
[2025-05-21T09:30:26.807+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.fetchRegionFromCache(AmazonS3Client.java:6162)
[2025-05-21T09:30:26.808+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5211)
[2025-05-21T09:30:26.809+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5173)
[2025-05-21T09:30:26.809+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5167)
[2025-05-21T09:30:26.809+0000] {subprocess.py:93} INFO - 	at com.amazonaws.services.s3.AmazonS3Client.listObjectsV2(AmazonS3Client.java:963)
[2025-05-21T09:30:26.810+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AFileSystem.lambda$listObjects$7(S3AFileSystem.java:2116)
[2025-05-21T09:30:26.810+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.statistics.impl.IOStatisticsBinding.lambda$trackDurationOfOperation$5(IOStatisticsBinding.java:499)
[2025-05-21T09:30:26.811+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.Invoker.retryUntranslated(Invoker.java:412)
[2025-05-21T09:30:26.811+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.Invoker.retryUntranslated(Invoker.java:375)
[2025-05-21T09:30:26.812+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AFileSystem.listObjects(S3AFileSystem.java:2107)
[2025-05-21T09:30:26.812+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.S3AFileSystem$ListingOperationCallbacksImpl.lambda$listObjectsAsync$0(S3AFileSystem.java:1750)
[2025-05-21T09:30:26.813+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.impl.CallableSupplier.get(CallableSupplier.java:62)
[2025-05-21T09:30:26.815+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
[2025-05-21T09:30:26.816+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[2025-05-21T09:30:26.816+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[2025-05-21T09:30:26.817+0000] {subprocess.py:93} INFO - 	at java.base/java.lang.Thread.run(Thread.java:829)
[2025-05-21T09:30:26.817+0000] {subprocess.py:93} INFO - Caused by: com.amazonaws.SdkClientException: Unable to load AWS credentials from environment variables (AWS_ACCESS_KEY_ID (or AWS_ACCESS_KEY) and AWS_SECRET_KEY (or AWS_SECRET_ACCESS_KEY))
[2025-05-21T09:30:26.817+0000] {subprocess.py:93} INFO - 	at com.amazonaws.auth.EnvironmentVariableCredentialsProvider.getCredentials(EnvironmentVariableCredentialsProvider.java:50)
[2025-05-21T09:30:26.817+0000] {subprocess.py:93} INFO - 	at org.apache.hadoop.fs.s3a.AWSCredentialProviderList.getCredentials(AWSCredentialProviderList.java:177)
[2025-05-21T09:30:26.820+0000] {subprocess.py:93} INFO - 	... 27 more
[2025-05-21T09:30:26.821+0000] {subprocess.py:93} INFO - 
[2025-05-21T09:30:26.822+0000] {subprocess.py:93} INFO - 25/05/21 09:30:26 INFO SparkUI: Stopped Spark web UI at http://c4c55d29c79f:4041
[2025-05-21T09:30:26.822+0000] {subprocess.py:93} INFO - 25/05/21 09:30:26 INFO StandaloneSchedulerBackend: Shutting down all executors
[2025-05-21T09:30:26.823+0000] {subprocess.py:93} INFO - 25/05/21 09:30:26 INFO CoarseGrainedSchedulerBackend$DriverEndpoint: Asking each executor to shut down
[2025-05-21T09:30:26.854+0000] {subprocess.py:93} INFO - 25/05/21 09:30:26 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
[2025-05-21T09:30:26.879+0000] {subprocess.py:93} INFO - 25/05/21 09:30:26 INFO MemoryStore: MemoryStore cleared
[2025-05-21T09:30:26.880+0000] {subprocess.py:93} INFO - 25/05/21 09:30:26 INFO BlockManager: BlockManager stopped
[2025-05-21T09:30:26.888+0000] {subprocess.py:93} INFO - 25/05/21 09:30:26 INFO BlockManagerMaster: BlockManagerMaster stopped
[2025-05-21T09:30:26.891+0000] {subprocess.py:93} INFO - 25/05/21 09:30:26 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
[2025-05-21T09:30:26.934+0000] {subprocess.py:93} INFO - 25/05/21 09:30:26 INFO SparkContext: Successfully stopped SparkContext
[2025-05-21T09:30:27.373+0000] {subprocess.py:93} INFO - 25/05/21 09:30:27 INFO ShutdownHookManager: Shutdown hook called
[2025-05-21T09:30:27.374+0000] {subprocess.py:93} INFO - 25/05/21 09:30:27 INFO ShutdownHookManager: Deleting directory /tmp/spark-39f06296-48ba-446f-ae64-ef1f9fabfa4e
[2025-05-21T09:30:27.385+0000] {subprocess.py:93} INFO - 25/05/21 09:30:27 INFO ShutdownHookManager: Deleting directory /tmp/spark-2a08584d-35bf-49ac-8cc6-532e2dd76a06
[2025-05-21T09:30:27.395+0000] {subprocess.py:93} INFO - 25/05/21 09:30:27 INFO ShutdownHookManager: Deleting directory /tmp/spark-39f06296-48ba-446f-ae64-ef1f9fabfa4e/pyspark-752b805a-744b-454e-84ee-9300b4e5bf0c
[2025-05-21T09:30:27.403+0000] {subprocess.py:93} INFO - 25/05/21 09:30:27 INFO MetricsSystemImpl: Stopping s3a-file-system metrics system...
[2025-05-21T09:30:27.404+0000] {subprocess.py:93} INFO - 25/05/21 09:30:27 INFO MetricsSystemImpl: s3a-file-system metrics system stopped.
[2025-05-21T09:30:27.405+0000] {subprocess.py:93} INFO - 25/05/21 09:30:27 INFO MetricsSystemImpl: s3a-file-system metrics system shutdown complete.
[2025-05-21T09:30:27.460+0000] {subprocess.py:97} INFO - Command exited with return code 0
[2025-05-21T09:30:27.504+0000] {taskinstance.py:1398} INFO - Marking task as SUCCESS. dag_id=yellow_taxi_batch_pipeline, task_id=transform_taxi_data, execution_date=20250521T092724, start_date=20250521T093000, end_date=20250521T093027
[2025-05-21T09:30:27.565+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 0
[2025-05-21T09:30:27.591+0000] {taskinstance.py:2776} INFO - 0 downstream tasks scheduled from follow-on schedule check
